endpoints/decorators/metrics.go

```diff
@@ -49,7 +62,17 @@ func (w *writerWithStatus) WriteHeader(statusCode int) {
 	w.delegate.WriteHeader(statusCode)
 }
 
+// Write is writerWithStatus's implementation of the http.ResponseWriter interface
+// Write(bytes []byte) (int, error) method and includes logic to set the wasKeyProvided
+// field to true if the "IsKeyProvided" keyword is passed as parameter. If not, bytes
+// is assumed to be the Put response and is passed as parameter to http.ResponseWriter's
+// Write(bytes []byte) (int, error) instead.
 func (w *writerWithStatus) Write(bytes []byte) (int, error) {
+	if string(bytes) == CustomUuidKey {
+		w.wasKeyProvided = true
+		return 0, nil
+	}
```

Comment:
This still looks a hacky. I also think this violates the intention of decoupling the metrics from the handlers, as the handler now has to reference this magic value. There may simple be a need for the PUT endpoint to have a reference to the metrics to mark when this occurs, because the decorators are unable to figure it own on their own.

Reply:
This problem has been solved in recently merged PR #89. Let me know if you agree.

Reply:
I thought we agreed to move the `CustomUuidKey` hack and move the call to the new key provided metric directly in the put endpoint handler?

Reply:
Sorry I forgot to rollback this file. Done.